.github/workflows/tests.yaml

```diff
@@ -0,0 +1,45 @@
+name: Tests
+on:
+  workflow_call:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+  integration-test-microk8s:
+    name: Integration tests (microk8s)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+      - name: Run integration tests
+        # set a predictable model name so it can be consumed by charm-logdump-action
+        run: tox -e integration -- --model testing
```

Comment:
Iirc, we had trouble with this and we no longer use it that way:
- itests fail because attempting to create a new model "testing" while the old one is still destroying
- charm-logdump-action needs some love regardless

I think this would work for cases where there is only one *.py itest file.

Reply:
Maybe we should refactor the logdump action to just dump all logs from all models, and then expect people to parse it afterwards.  It is already a situation where we're in a crash recovery state, so doing a little extra parsing might be ok.  thoughts?

Reply:
Yes, I was thinking of the same, especially as an immediate solution.
But probably `modeloperator` could be omitted.
Also I'd imagine there is some kubectl jsonpath magic that would let you obtain the most recently active namespace.